### PR TITLE
sysdig-cli-scanner: init at 1.13.2

### PR DESCRIPTION
**Pull request by tembleking**
## Description of changes

The Sysdig Vulnerability CLI Scanner, [sysdig-cli-scanner](https://docs.sysdig.com/en/docs/installation/sysdig-secure/install-vulnerability-cli-scanner/), is a versatile tool designed to manually scan container images and directories, whether they are located locally or remotely. Depending on your specific use case, you have the flexibility to execute sysdig-cli-scanner in Vulnerability Management (VM) mode for image scanning or Infrastructure as Code (IaC) mode for scanning directories

Latest version: 1.13.2 (Can be checked at: https://download.sysdig.com/scanning/sysdig-cli-scanner/latest_version.txt)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/sy/sysdig-cli-scanner/package.nix
Comment (afh) [inline pkgs/by-name/sy/sysdig-cli-scanner/package.nix:33]: According to the [recommendations for meta attributes](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#meta-attributes) description should
* not start with an article
* avoid subjective language
```suggestion
    description = "Tool for scanning container images and directories using Sysdig";
```

File: pkgs/by-name/sy/sysdig-cli-scanner/package.nix
Comment (afh) [inline pkgs/by-name/sy/sysdig-cli-scanner/package.nix:50]: Since this package isn't built from source its source provenance should be declared accordingly
```suggestion
    ];
    sourceProvenance = with sourceTypes; [ binaryNativeCode ];
```

File: pkgs/by-name/sy/sysdig-cli-scanner/package.nix
Comment (afh) [inline pkgs/by-name/sy/sysdig-cli-scanner/package.nix:53]: In all of the nixpkgs I've come across the `meta` section always came last. What's the motivation to place `passthru` last? What other package you know of do this too?

File: pkgs/by-name/sy/sysdig-cli-scanner/package.nix
Comment (afh) [inline pkgs/by-name/sy/sysdig-cli-scanner/package.nix:9]: Haven't come across this pattern; very interesting approach!

File: pkgs/by-name/sy/sysdig-cli-scanner/update.sh
Comment (afh) [inline pkgs/by-name/sy/sysdig-cli-scanner/update.sh:47]: It seems easy enough to generate valid nix code here to avoid the need to convert from toml to nix. What are the advantages of using toml here?

File: pkgs/by-name/sy/sysdig-cli-scanner/update.sh
Comment (afh) [inline pkgs/by-name/sy/sysdig-cli-scanner/update.sh:2]: Out of curiosity: which part of update.sh needs coreutils-full?

**Review by afh — CHANGES_REQUESTED** [review]
Thanks for this, @tembleking.

Please find below a few questions and some suggestions for improvement.

File: pkgs/by-name/sy/sysdig-cli-scanner/package.nix
Comment (tembleking) [reply pkgs/by-name/sy/sysdig-cli-scanner/package.nix:33]: Thanks for the information @afh, I will change this with your suggestion and I will squash all the commits in the PR so it's cleaner.

File: pkgs/by-name/sy/sysdig-cli-scanner/package.nix
Comment (tembleking) [reply pkgs/by-name/sy/sysdig-cli-scanner/package.nix:50]: I didn't know that! Thanks for the hint.

File: pkgs/by-name/sy/sysdig-cli-scanner/package.nix
Comment (tembleking) [reply pkgs/by-name/sy/sysdig-cli-scanner/package.nix:53]: No particular motivation, really, I will put it before the `meta` section. :+1:

File: pkgs/by-name/sy/sysdig-cli-scanner/package.nix
Comment (tembleking) [reply pkgs/by-name/sy/sysdig-cli-scanner/package.nix:9]: Thank you!

File: pkgs/by-name/sy/sysdig-cli-scanner/update.sh
Comment (tembleking) [reply pkgs/by-name/sy/sysdig-cli-scanner/update.sh:47]: I can try generating Nix instead of TOML :+1:

File: pkgs/by-name/sy/sysdig-cli-scanner/update.sh
Comment (tembleking) [reply pkgs/by-name/sy/sysdig-cli-scanner/update.sh:2]: I really put it just in case, I don't actually know if all the commands would work without it. I can remove it if you feel like it's redundant.

Comment (afh) [reply pkgs/by-name/sy/sysdig-cli-scanner/update.sh]: @tembleking, I'm not too familiar with the resources that update scripts take, so I have no authority on this. That said, I'd go for the smallest possible packages, especially as `coreutils` is used in ~30 update scripts, but `coreutils-full` isn't; at least as far as my quick test shows:

```sh
nix run nixpkgs#silver-searcher -- -0g update.sh | xargs -0 -n1 sed -n 2p | nix run nixpkgs#silver-searcher -- coreutils | nl
     1  #!nix-shell -I nixpkgs=./. -i bash -p jq curl common-updater-scripts nix coreutils
     2  #! nix-shell -i bash -p gnugrep gnused coreutils curl wget jq nix-update prefetch-npm-deps nodejs
     3  #!nix-shell -i bash -p curl jq coreutils common-updater-scripts
# output truncated for demonstration purposes…
```

File: pkgs/by-name/sy/sysdig-cli-scanner/update.sh
Comment (afh) [reply pkgs/by-name/sy/sysdig-cli-scanner/update.sh]: If you think that's interesting, alright, but I currently have no good reason for recommending this. Maybe other nixpkgs folks can chime in on this…

**Review by afh — COMMENTED** [review]
Thanks for accepting the suggestions. I haven't had too much experience with update scripts, so I've only taken a rudimentary review pass on it.
Generally this LGTM, yet I'd appreciate if seasoned nixpkgs reviewers would review this too.

@AndersonTorres, @SuperSandro2000, @wegank, would you be interested in lending your expertise to this PR?

**Comment by tembleking** [comment]
Adding @AndersonTorres, @SuperSandro2000 and @wegank as reviewers just so they can view the PR :)

**Comment by wegank** [comment]
LGTM overall, but it seems that `nix-shell maintainers/scripts/update.nix --argstr package sysdig-cli-scanner` generates a `sysdig-cli-scanner.versions.nix` in the wrong directory?

File: pkgs/by-name/sy/sysdig-cli-scanner/update.sh
Comment (afh) [inline pkgs/by-name/sy/sysdig-cli-scanner/update.sh:19]: This could well be a matter of personal preference, yet I'd pass the versions file as argument to `appendToVersionsFile` instead of having the function use the global variable.
```suggestion
      appendToVersionsFile "$VERSIONS_FILE" "$arch" "$os" "$download_url" "$file_hash"
```

File: pkgs/by-name/sy/sysdig-cli-scanner/update.sh
Comment (afh) [inline pkgs/by-name/sy/sysdig-cli-scanner/update.sh:44]: See previous comment:
```suggestion
  local file="$1"
  local architecture="$2"
  local operating_system="$3"
  local url="$4"
  local hash="$5"
  cat >> "$file" << EOF
```

File: pkgs/by-name/sy/sysdig-cli-scanner/update.sh
Comment (afh) [inline pkgs/by-name/sy/sysdig-cli-scanner/update.sh:9]: The following should writh the versions file to the correct filepath:
```suggestion
SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
VERSIONS_FILE="${SCRIPT_DIR}/sysdig-cli-scanner.versions.nix"
```

**Review by afh — COMMENTED** [review]
Thanks for chiming in on this @wegank, very much appreciated 🙏 

@tembleking please find below a few ideas for this PR…

File: pkgs/by-name/sy/sysdig-cli-scanner/package.nix
Comment (SuperSandro2000) [inline pkgs/by-name/sy/sysdig-cli-scanner/package.nix:27]: Do we really want to use a predictable directory for that?
Also people using a tmpfs under /tmp won't benefit from this.

Comment (tembleking) [reply pkgs/by-name/sy/sysdig-cli-scanner/package.nix:27]: If `dbpath` is not specified, it tries to use the same path of the binary, which is the path under `/nix/store`, which is not writable, and therefore it fails to run.

Users using a tmpfs under `/tmp` should not have any problem, the DB is updated every 12 hours so it has to be replaced anyways after some time. I can modify it to use `~/.cache/sysdig-cli-scanner` instead if you think that's better.

File: pkgs/by-name/sy/sysdig-cli-scanner/update.sh
Comment (tembleking) [reply pkgs/by-name/sy/sysdig-cli-scanner/update.sh:19]: Sounds very good, seems a good change to have.

**Comment by tembleking** [comment]
> Thanks for chiming in on this @wegank, very much appreciated 🙏

Indeed, thank you so much for the feedback @wegank 

> @tembleking please find below a few ideas for this PR…
Thank you @afh for the changes, I just applied them. I will rebase and squash the commits so it's cleaner in the PR. :100:

File: pkgs/by-name/sy/sysdig-cli-scanner/package.nix
Comment (afh) [reply pkgs/by-name/sy/sysdig-cli-scanner/package.nix:27]: +1 on using `~/.cache`

Comment (tembleking) [reply pkgs/by-name/sy/sysdig-cli-scanner/package.nix:27]: Unfortunately, after some tests I did, if you specify 

```
    wrapProgram $out/bin/sysdig-cli-scanner \
      --add-flags --dbpath="~/.cache/sysdig-cli-scanner/"
```

The result is that a `~/.cache` is created in the current execution path, not in the home.

<details>
<summary>
See more details here:
</summary>
<pre><code>❯ nixpkgs-review rev HEAD

$ git -c fetch.prune=false fetch --no-tags --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0
remote: Enumerating objects: 4224, done.
remote: Counting objects: 100% (2441/2441), done.
remote: Compressing objects: 100% (183/183), done.
remote: Total 4224 (delta 2308), reused 2326 (delta 2242), pack-reused 1783
Receiving objects: 100% (4224/4224), 2.87 MiB | 3.23 MiB/s, done.
Resolving deltas: 100% (2904/2904), completed with 743 local objects.
From https://github.com/NixOS/nixpkgs
   de7067cf2f59..e9c63d27ca2d  master     -> refs/nixpkgs-review/0
$ git worktree add /home/sysdig/.cache/nixpkgs-review/rev-9be1a0dcd204b16534f5aaa93d188322afb1979d/nixpkgs e9c63d27ca2d4ec12b96d21393a00aa3cdda40aa
Preparing worktree (detached HEAD e9c63d27ca2d)
Updating files: 100% (42758/42758), done.
HEAD is now at e9c63d27ca2d Merge pull request #323988 from NyCodeGHG/gauge-patchelf
$ nix-env --extra-experimental-features no-url-literals --option system x86_64-linux -f <nixpkgs> --nix-path nixpkgs=/home/sysdig/.cache/nixpkgs-review/rev-9be1a0dcd204b16534f5aaa93d188322afb1979d/nixpkgs nixpkgs-overlays=/tmp/tmpenkhg6v8 -qaP --xml --out-path --show-trace --no-allow-import-from-derivation
$ git merge --no-commit --no-ff 9be1a0dcd204b16534f5aaa93d188322afb1979d
Automatic merge went well; stopped before committing as requested
$ nix-env --extra-experimental-features no-url-literals --option system x86_64-linux -f <nixpkgs> --nix-path nixpkgs=/home/sysdig/.cache/nixpkgs-review/rev-9be1a0dcd204b16534f5aaa93d188322afb1979d/nixpkgs nixpkgs-overlays=/tmp/tmpenkhg6v8 -qaP --xml --out-path --show-trace --no-allow-import-from-derivation --meta
1 package added:
sysdig-cli-scanner (init at 1.13.2)

$ nix build --file /nix/store/ww9sz3m14wsh484vchhgws9vhg2mlfcx-nixpkgs-review-2.10.4/lib/python3.11/site-packages/nixpkgs_review/nix/review-shell.nix --nix-path nixpkgs=/home/sysdig/.cache/nixpkgs-review/rev-9be1a0dcd204b16534f5aaa93d188322afb1979d/nixpkgs nixpkgs-overlays=/tmp/tmpenkhg6v8 --extra-experimental-features nix-command no-url-literals --no-link --keep-going --no-allow-import-from-derivation --option build-use-sandbox relaxed --argstr system x86_64-linux --argstr nixpkgs-path /home/sysdig/.cache/nixpkgs-review/rev-9be1a0dcd204b16534f5aaa93d188322afb1979d/nixpkgs --argstr nixpkgs-config-path /tmp/tmpl6dqpb7m.nix --argstr attrs-path /home/sysdig/.cache/nixpkgs-review/rev-9be1a0dcd204b16534f5aaa93d188322afb1979d/attrs.nix
1 package built:
sysdig-cli-scanner

$ /nix/store/wxp968584yl2ygfm3kwhl712wc2y6r5k-nix-2.18.5/bin/nix-shell --argstr system x86_64-linux --argstr nixpkgs-path /home/sysdig/.cache/nixpkgs-review/rev-9be1a0dcd204b16534f5aaa93d188322afb1979d/nixpkgs --argstr nixpkgs-config-path /tmp/tmpl6dqpb7m.nix --argstr attrs-path /home/sysdig/.cache/nixpkgs-review/rev-9be1a0dcd204b16534f5aaa93d188322afb1979d/attrs.nix --nix-path nixpkgs=/home/sysdig/.cache/nixpkgs-review/rev-9be1a0dcd204b16534f5aaa93d188322afb1979d/nixpkgs nixpkgs-overlays=/tmp/tmpenkhg6v8 /nix/store/ww9sz3m14wsh484vchhgws9vhg2mlfcx-nixpkgs-review-2.10.4/lib/python3.11/site-packages/nixpkgs_review/nix/review-shell.nix

❯ sysdig-cli-scanner -a https://us2.app.sysdig.com/ quay.io/sysdig/cloud-connector
2024-08-09T09:38:29+02:00 Starting analysis with Sysdig scanner version 1.13.2
2024-08-09T09:38:30+02:00 Retrieving MainDB...
2024-08-09T09:42:38+02:00 Done 167.8 MB
2024-08-09T09:42:39+02:00 Loading MainDB...
2024-08-09T09:42:39+02:00 Done
2024-08-09T09:42:40+02:00 Retrieving image...
INFO[0250] overlay: test mount did not indicate whether or not metacopy is being used: failed to mount overlay for metacopy check with "" options: operation not permitted 
2024-08-09T09:42:52+02:00 Done
2024-08-09T09:42:52+02:00 Scan started...
2024-08-09T09:42:54+02:00 Uploading image sbom to backend...
2024-08-09T09:42:54+02:00 Uploading scanresult with sbom to backend...
2024-08-09T09:42:56+02:00 Done
2024-08-09T09:42:56+02:00 Total execution time 4m26.999835227s

Type: dockerImage
ImageID: sha256:7f8b0cc914659812ddab3994880b5b9b75c20a0ea4b1fbebbd94f79c3091c34b
Digest: sha256:b39c308f67c4f6984fc0e2423f5dc4ae1d7c9131d1074bf6e63d563a4d37fdae
BaseOS: redhat 8.10
PullString: quay.io/sysdig/cloud-connector

30 vulnerabilities found
0 Critical (0 fixable)
0 High (0 fixable)
1 Medium (0 fixable)
26 Low (0 fixable)
3 Negligible (0 fixable)

                                    POLICIES EVALUATION
    Policy: Critical Vulnerability Found          PASSED (0 failures - 0 risks accepted)
    Policy: Sensitive Information or Secret Found PASSED (0 failures - 0 risks accepted)

Policies evaluation PASSED at 2024-08-09T09:42:56+02:00
Full image results here: https://us2.app.sysdig.com/secure/#/vulnerabilities/results/17e9ff9b48d8eb6fe886aff929a97a30/overview (id 17e9ff9b48d8eb6fe886aff929a97a30)
Execution logs written to: /home/sysdig/.cache/nixpkgs-review/rev-9be1a0dcd204b16534f5aaa93d188322afb1979d/scan-logs

❯ ls -al
total 20
drwxr-xr-x 1 sysdig users  114 ago  9 09:38  .
drwxr-xr-x 1 sysdig users  268 ago  9 09:32  ..
drwxr-xr-x 1 sysdig users   12 ago  9 09:38 '~'
-rw-r--r-- 1 sysdig users   41 ago  9 09:37  attrs.nix
drwxr-xr-x 1 sysdig users   44 ago  9 09:37  logs
drwxr-xr-x 1 sysdig users  348 ago  9 09:32  nixpkgs
-rw-r--r-- 1 sysdig users  233 ago  9 09:37  report.json
-rw-r--r-- 1 sysdig users  198 ago  9 09:37  report.md
drwxr-xr-x 1 sysdig users   36 ago  9 09:37  results
-rw-r--r-- 1 sysdig users 4900 ago  9 09:42  scan-logs</code></pre>
</details>

See the `drwxr-xr-x 1 sysdig users   12 ago  9 09:38 '~'` line. Unfortunately, I think we will need to leave it to `/tmp`, unless you know how to reference the actual home.

Comment (afh) [reply pkgs/by-name/sy/sysdig-cli-scanner/package.nix:27]: How about using `$HOME` (might need to be escaped `\$HOME`) instead of `~`? Does that work, @tembleking?

Comment (tembleking) [reply pkgs/by-name/sy/sysdig-cli-scanner/package.nix]: That one works! I am commiting it now, thanks for the suggestion!

Comment (tembleking) [reply pkgs/by-name/sy/sysdig-cli-scanner/package.nix]: What do you think about the change folks? @afh @SuperSandro2000

File: pkgs/by-name/sy/sysdig-cli-scanner/package.nix
Comment (afh) [inline pkgs/by-name/sy/sysdig-cli-scanner/package.nix:24]: I think being a bit more specific here could be helpful:
```suggestion
    install -Dm755 -T $src $out/bin/sysdig-cli-scanner
```

**Review by afh — APPROVED** [review]
Apologies for not being more responsive on this, @tembleking, my attention was needed elsewhere.

Please find below a minor, optional suggestion for improvement and otherwise LGTM

File: pkgs/by-name/sy/sysdig-cli-scanner/package.nix
Comment (SuperSandro2000) [reply pkgs/by-name/sy/sysdig-cli-scanner/package.nix]: Using ~/.cache for this feels a lot nicer

**Comment by afh** [comment]
One more thing, @tembleking, could you squash the commits into a single one please.

File: pkgs/by-name/sy/sysdig-cli-scanner/package.nix
Comment (tembleking) [reply pkgs/by-name/sy/sysdig-cli-scanner/package.nix]: Sure, makes sense!

**Comment by tembleking** [comment]
> One more thing, @tembleking, could you squash the commits into a single one please.

Thanks for the suggestion, I squashed all the commits into a single one. Let me know if you'd change anything else!

**Comment by afh** [comment]
Thanks for the extra effort, @tembleking, much appreciated.
LGTM 👍

**Comment by afh** [comment]
Thanks for merging this, @wegank, much appreciated 👍

**Comment by tembleking** [comment]
> Thanks for merging this, @wegank, much appreciated 👍

Indeed, thank you @wegank 💯

**Pull request merged**
